tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -60,12 +60,23 @@ QualType getUintTypeWithSourceComponents(const ASTContext &astContext,
                    "getUintTypeWithSourceComponents");
 }
 
-uint32_t getLocationCount(const ASTContext &astContext, QualType type) {
+uint32_t getLocationCount(const ASTContext &astContext, QualType type,
+                          bool rowMajorAttr) {
   // See Vulkan spec 14.1.4. Location Assignment for the complete set of rules.
 
+  // Check if it has a matrix majorness attribute before getting the canonical
+  // type. If yes and it is row_major attribute, we set `rowMajorAttr` as true.
+  llvm::Optional<bool> isRowMajorAttr;
+  if (containsMatrixMajorAttr(astContext, type, &isRowMajorAttr)) {
```

Comment:
it's better to update the `isRowMajorMatrix` function and call that one here. If `isRowMajorMatrix` is called, there's no need for llvm::Optional variable.